### PR TITLE
[ShazamKit] Add support for Xcode 15

### PR DESCRIPTION
**Pull request by haritha-mohan**
WatchOS todo/xtro was weird, so added it to the ignore for now..all intro tests look good though

### CODE REVIEW

File: src/shazamkit.cs [hunk 1]
```diff
@@ -23,6 +23,7 @@ public enum SHErrorCode : long {
 		CustomCatalogInvalidURL = 301,
 		MediaLibrarySyncFailed = 400,
 		InternalError = 500,
+		MediaItemFetchFailed = 600
```
Comment (mandel-macaque) [inline src/shazamkit.cs:26]:
```suggestion
		MediaItemFetchFailed = 600,
```

**Review by mandel-macaque — APPROVED** [review]
Approved. I committed a small nit.

File: src/shazamkit.cs [hunk 2]
```diff
@@ -62,6 +63,9 @@ enum SHMediaItemProperty {
 		[Watch (9, 0), TV (16, 0), Mac (13, 0), iOS (16, 0), MacCatalyst (16, 0)]
 		[Field ("SHMediaItemFrequencySkewRanges")]
 		FrequencySkewRanges,
+		[Watch (10, 0), TV (17, 0), Mac (14, 0), iOS (17, 0), MacCatalyst (17, 0)]
+		[Field ("SHMediaItemCreationDate")]
+		SHMediaItemCreationDate,
```
Comment (rolfbjarne) [inline src/shazamkit.cs:68]:
```suggestion
		CreationDate,
```

File: tests/xtro-sharpie/watchOS-ShazamKit.ignore [hunk 3]
```diff
@@ -27,3 +28,4 @@
 !unknown-type! SHSession bound
 !unknown-type! SHSignature bound
 !unknown-type! SHSignatureGenerator bound
+!unknown-field! SHMediaItemCreationDate bound
```
Comment (rolfbjarne) [inline tests/xtro-sharpie/watchOS-ShazamKit.ignore:31]: Missing EOL:

```suggestion
!unknown-field! SHMediaItemCreationDate bound

```

File: src/shazamkit.cs [hunk 4]
```diff
@@ -190,9 +194,17 @@ interface SHMediaItem : NSSecureCoding, NSCopying {
...
+		[Watch (10, 0), TV (17, 0), Mac (14, 0), iOS (17, 0), MacCatalyst (17, 0)]
+		[NullAllowed, Export ("creationDate", ArgumentSemantic.Strong)]
+		NSDate CreationDate { get; }
 	}
 
 	[iOS (15, 0), Mac (12, 0), Watch (8, 0), TV (15, 0), MacCatalyst (15, 0)]
+	[Deprecated (PlatformName.MacOSX, 12, 0, message: "Use SHLibrary instead.")]
+	[Deprecated (PlatformName.iOS, 15, 0, message: "Use SHLibrary instead.")]
+	[Deprecated (PlatformName.TvOS, 15, 0, message: "Use SHLibrary instead.")]
+	[Deprecated (PlatformName.WatchOS, 8, 0, message: "Use SHLibrary instead.")]
```
Comment (rolfbjarne) [inline src/shazamkit.cs:207]:
```suggestion
	[Deprecated (PlatformName.WatchOS, 8, 0, message: "Use SHLibrary instead.")]
	[Deprecated (PlatformName.MacCatalyst, 15, 0, message: "Use SHLibrary instead.")]
```

**Review by rolfbjarne — CHANGES_REQUESTED** [review]

**Review by rolfbjarne — APPROVED** [review]

**Review by dalexsoto — APPROVED** [review]

**Review by rolfbjarne — APPROVED** [review]

**Comment by mandel-macaque** [comment]
@haritha-mohan  I have requested a rerun of the failing tests.

**Pull request merged**